packages/composer/lib/proxy.js

```diff
@@ -81,33 +81,15 @@ module.exports = fp(async function (app, opts) {
     const basePath = `/${prefix ?? ''}`.replaceAll(/\/+/g, '/').replace(/\/$/, '')
     const dispatcher = getGlobalDispatcher()
 
-    if (needsRootRedirect) {
-      app.addHook('preHandler', (req, reply, done) => {
-        if (req.url === basePath) {
-          app.inject(
-            {
-              method: req.method,
-              url: `${basePath}/`,
-              headers: req.headers,
-              payload: req.body
-            },
-            (err, result) => {
-              if (err) {
-                done(err)
-                return
-              }
-
-              const replyHeaders = result.headers
-              delete replyHeaders['content-length']
-              delete replyHeaders['transfer-encoding']
-
-              reply.code(result.statusCode).headers(replyHeaders).send(result.rawPayload)
-              done()
-            }
-          )
-        } else {
+    if (needsRootTrailingSlash) {
+      app.addHook('preHandler', function rootTrailingSlashPreHandler (req, reply, done) {
+        if (req.url !== basePath) {
```

Comment:
Instead of adding a `preHanlder, I think you can just add a new route.

Reply:
composer+node. But I guess it's generic.